httpclient/sentryhttpclient.go

```diff
@@ -0,0 +1,102 @@
+// Package sentryhttpclient provides Sentry integration for Requests modules to enable distributed tracing between services.
+// It is compatible with `net/http.RoundTripper`.
+//
+//	import sentryhttpclient "github.com/getsentry/sentry-go/httpclient"
+//
+//	roundTrippper := sentryhttpclient.NewSentryRoundTripper(nil, nil)
+//	client := &http.Client{
+//		Transport: roundTripper,
+//	}
+//
+//	request, err := client.Do(request)
+package sentryhttpclient
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/getsentry/sentry-go"
+)
+
+// SentryRoundTripTracerOption provides a specific type in which defines the option for SentryRoundTripper.
+type SentryRoundTripTracerOption func(*SentryRoundTripper)
+
+// WithTags allows the RoundTripper to includes additional tags.
+func WithTags(tags map[string]string) SentryRoundTripTracerOption {
+	return func(t *SentryRoundTripper) {
+		for k, v := range tags {
+			t.tags[k] = v
+		}
+	}
+}
+
+// WithTag allows the RoundTripper to includes additional tag.
+func WithTag(key, value string) SentryRoundTripTracerOption {
+	return func(t *SentryRoundTripper) {
+		t.tags[key] = value
+	}
+}
+
+// NewSentryRoundTripper provides a wrapper to existing http.RoundTripper to have required span data and trace headers for outgoing HTTP requests.
+//
+//   - If `nil` is passed to `originalRoundTripper`, it will use http.DefaultTransport instead.
+func NewSentryRoundTripper(originalRoundTripper http.RoundTripper, opts ...SentryRoundTripTracerOption) http.RoundTripper {
+	if originalRoundTripper == nil {
+		originalRoundTripper = http.DefaultTransport
+	}
+
+	t := &SentryRoundTripper{
+		originalRoundTripper: originalRoundTripper,
+		tags:                 make(map[string]string),
+	}
+
+	for _, opt := range opts {
+		if opt != nil {
+			opt(t)
+		}
+	}
+
+	return t
+}
+
+// SentryRoundTripper provides a http.RoundTripper implementation for Sentry Requests module.
+type SentryRoundTripper struct {
+	originalRoundTripper http.RoundTripper
+
+	tags map[string]string
+}
+
+func (s *SentryRoundTripper) RoundTrip(request *http.Request) (*http.Response, error) {
+	ctx := request.Context()
+	cleanRequestURL := request.URL.Redacted()
+
+	span := sentry.StartSpan(ctx, "http.client", sentry.WithTransactionName(fmt.Sprintf("%s %s", request.Method, cleanRequestURL)))
+	span.Tags = s.tags
+	defer span.Finish()
```

Comment:
We do not want to always start transactions here. If one already exists, start a child span.

Reply:
Just to be clear: http.client spans should only created if there is any parent span?

Reply:
Yes, Sentry does not work well with these kind of single span transactions. We can revisit this once span streaming is fully supported in the product, but this will take a few months.

Reply:
https://sentry.io/insights/http/ should also work with these spans fwiw, docs are [here](https://develop.sentry.dev/sdk/telemetry/traces/modules/requests/).

Reply:
*which you already mentioned in the PR description, nvm

Reply:
Another question (unrelated): Does that also affects all the span op within the insights module? Meaning not only `http.client`, but `db.sql` and `messaging.*` also counts?

Reply:
I guess technically we extract all the required fields from a transaction and store it into the span data set, though this is not how these modules where initially designed to work.